Library/Homebrew/formula.rb

```diff
@@ -1671,6 +1672,13 @@ def self.core_alias_reverse_table
     CoreTap.instance.alias_reverse_table
   end
 
+  # Returns a list of approximately matching formula names, but not the complete match
+  # @private
+  def self.fuzzy_search(name)
+    @spell_checker ||= ::DidYouMean::SpellChecker.new(dictionary: names)
```

Comment:
May also be worth including `full_names` in here. Similarly, may want to bust this cache when a new tap is tapped.